lectures/sympy.md

```diff
@@ -0,0 +1,924 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.5
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+```{code-cell} ipython3
+!pip install sympy==1.12
+```
+
++++ {"user_expressions": []}
+
+# SymPy
+
+## Overview
+
+Unlike numerical libraries that deal with values, [SymPy](https://www.sympy.org/en/index.html) focuses on manipulating mathematical symbols and expressions directly.
+
+SymPy provides [a wide range of functionalities](https://www.sympy.org/en/features.html) including 
+
+- Symbolic Expression
+- Equation Solving
+- Simplification
+- Calculus
+- Matrices
+- Discrete Math, etc.
+
+These functions make SymPy a popular open-source alternative to other proprietary symbolic computation software such as Mathematica.
+
+In this lecture, we will explore some of the functionalities of SymPy and demonstrate how to use basic SymPy functions to solve economic models.
+
+## Getting Started
+
+Let’s first import the library and initialize the printer for symbolic output
+
+```{code-cell} ipython3
+from sympy import *
+from sympy.plotting import plot, plot3d_parametric_line, plot3d
+from sympy.solvers.inequalities import reduce_rational_inequalities
+from sympy.stats import Poisson, Exponential, Binomial, density, moment, E
+
+import numpy as np
+
+# Enable the best printer available
+init_printing()
+```
+
++++ {"user_expressions": []}
+
+## Symbolic algebra
+
+### Symbols
+
++++ {"user_expressions": []}
+
+Before we start manipulating the symbols, we initialize some symbols to work with
+
+```{code-cell} ipython3
+x, y, z = symbols('x y z')
+```
+
++++ {"user_expressions": []}
+
+Symbols are the basic units for symbolic computation in SymPy.
+
++++ {"user_expressions": []}
+
+### Expressions
+
+We can now use symbols `x`, `y`, and `z` to build expressions and equations.
+
+Here we build a simple expression first
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 10%
```

Comment:
@HumphreyYang @HengchengZhang should we add some rules to our manual around using symbolics?

Reply:
Hi @mmcky,

I just removed those headers, and the formula looks fine on my local build. @HengchengZhang, were these headers added because of the size of the formula?